src/qonnx/transformation/channelsLast.py

```diff
@@ -0,0 +1,460 @@
+import warnings
+from onnx import TensorProto, helper
+
+from finn.transformation.base import Transformation
+from finn.transformation.general import RemoveUnusedTensors
+from finn.transformation.infer_shapes import InferShapes
+from finn.util.basic import get_by_name
+from qonnx.custom_op import ChannelsLast
+
+# ToDo: These parameters also exist for the ChannelsLast wrapped_ops, somehow they should be moved to a shared location.
+# Standard ONNX nodes which require a ChannelsLast data format to function properly
+_channelsLast_node_types = list(ChannelsLast.custom_op.keys())
+# Required for ChannelsLast transformations and ops
+# Transpose parameters to convert to channels last for 3D and 4D tensors
+_to_chan_last_args = {
+    3: (0, 2, 1),
+    4: (0, 2, 3, 1),
+}
+# Similarly for converting back to channels first.
+_to_chan_first_args = {
+    3: (0, 2, 1),
+    4: (0, 3, 1, 2),
+}
+
+# Nodes, which do not modify the shape of the tensor
+# And modify all values in the same way.
+_move_through_nodes = ["Quant", "Relu"]
+
+# Nodes, which do not modify the shape of the tensor,
+# And modify all values in the same way, if the second tensor is a scalar.
+_move_through_nodes_if_scalar = ["Mul", "Div", "Sub", "Add"]
+
+
+class ConvertToChannelsLastAndClean(Transformation):
+    """
+    Converts data layout dependent nodes to ChannelsLast nodes and inserts transformations.
+    Then it tries to eliminate as many transformations as possible and moves the
+    still existing ones as far upstream as possible.
+    """
+
+    def apply(self, model):
+        assert model.check_all_tensor_shapes_specified(), "All tensor shapes should be specified."
+        model = model.transform(InsertChannelsLastDomainsAndTrafos())
+        max_tries = 10
+        for i in range(max_tries):
```

Comment:
make the 10 max_tries here an optional variable to the transformation's initializer function, stored as a member variable

ideally this should not be needed at all (see comment at the end of this function)